### DISCUSSION TITLE
Disabling repr support for some enums

### DISCUSSION THREAD

**Discussion by helaan** [juhaku/utoipa#831] [Q&A]
Hi there,

I'm running into issues where utoipa with the repr feature enabled is generating an openapi definition that is not compatible with the serde annotations I'm using. Because I'm using [sqlx's Type trait](https://docs.rs/sqlx/latest/sqlx/trait.Type.html#enumeration) I've added an #[repr(i8)] declaration, but I don't want to expose this enum directly to my API, so I derive Serialize/Deserialize for it. I did want to expose a few other enums, so these do use Serialize_repr/Deserialize_repr.

```rs
#[derive(sqlx::Type, Deserialize, Serialize, ToSchema)]
#[repr(i8)]
pub(crate) enum GameState {
	Unknown = -1,
	Reserved = 0,
	Lobby = 1,
	Started = 2,
	Ended = 3,
}
```

This generates a schema that suggests ints need to be used but the actual deserialization code expects strings.

Is it possible to disable Utoipa's repr detection on certain structs without disabling it on all structs?

**Comment by juhaku** [comment]
@helaan > Is it possible to disable Utoipa's repr detection on certain structs without disabling it on all structs?

Long story short, such feature has not been planned and only way around this is to do some duplicate work for now e.g. another enum that does not use `#[repr(i8)]`. 

Though if there are enough interest for this such attribute for `utoipa` could be added. E.g. `#[schema(skip_repr)]` Also pull requests are welcome.
